src/Dapr.Jobs/Models/DaprJobSchedule.cs

```diff
@@ -0,0 +1,143 @@
+    /// </summary>
+    private readonly Regex isEveryExpression = new(@"^@every (\d+(m?s|m|h))+$", RegexOptions.Compiled);
+    /// <summary>
+    /// The various prefixed period values allowed.
+    /// </summary>
+    private readonly string[] acceptablePeriodValues = { "yearly", "monthly", "weekly", "daily", "midnight", "hourly" };
+
+    /// <summary>
+    /// The value of the expression represented by the schedule.
+    /// </summary>
+    public string ExpressionValue { get; }
+
+    /// <summary>
+    /// Initializes the value of <see cref="ExpressionValue"/> based on the provided value from each of the factory methods.
+    /// </summary>
+    /// <remarks>
+    /// Developers are intended to create a new <see cref="DaprJobSchedule"/> using the provided static factory methods.
+    /// </remarks>
+    /// <param name="expressionValue">The value of the scheduling expression.</param>
+    internal DaprJobSchedule(string expressionValue)
+    {
+        ExpressionValue = expressionValue;
+    }
+
+    /// <summary>
+    /// Specifies a schedule built using the fluent Cron expression builder.
+    /// </summary>
+    /// <param name="builder">The fluent Cron expression builder.</param>
+    public static DaprJobSchedule FromCronExpression(CronExpressionBuilder builder)
+    {
+        ArgumentVerifier.ThrowIfNull(builder, nameof(builder));
+        return new DaprJobSchedule(builder.ToString());
+    }
+
+    /// <summary>
+    /// Specifies a single point in time.
+    /// </summary>
+    /// <param name="scheduledTime">The date and time when the job should be triggered.</param>
+    /// <returns></returns>
+    public static DaprJobSchedule FromDateTime(DateTimeOffset scheduledTime)
+    {
+        ArgumentVerifier.ThrowIfNull(scheduledTime, nameof(scheduledTime));
+        return new DaprJobSchedule(scheduledTime.ToString("O"));
+    }
+    
+    /// <summary>
+    /// Specifies a schedule using a Cron-like expression or '@' prefixed period strings.
+    /// </summary>
+    /// <param name="expression">The systemd Cron-like expression indicating when the job should be triggered.</param>
+    public static DaprJobSchedule FromExpression(string expression)
+    {
+        ArgumentVerifier.ThrowIfNullOrEmpty(expression, nameof(expression));
+        return new DaprJobSchedule(expression);
+    }
+
+    /// <summary>
+    /// Specifies a schedule using a duration interval articulated via a <see cref="TimeSpan"/>.
+    /// </summary>
+    /// <param name="duration">The duration interval.</param>
+    public static DaprJobSchedule FromDuration(TimeSpan duration)
+    {
+        ArgumentVerifier.ThrowIfNull(duration, nameof(duration));
+        return new DaprJobSchedule(duration.ToDurationString());
+    }
+
+    /// <summary>
+    /// Specifies a schedule in which the job is triggered to run once a year.
+    /// </summary>
+    public static DaprJobSchedule Yearly => new DaprJobSchedule("@yearly");
```

Comment:
Should these be `public static readonly` and assigned, rather than created on each call?

Reply:
I didn't see a point in having a backing field for each just for the readonly, so I gave them all getters and assigned the values once.